analysis_driver/transfer_data.py

```diff
@@ -139,7 +139,8 @@ def create_links_from_bcbio(sample_id, intput_dir, output_config, link_dir, quer
             links.append(link_file)
         else:
             app_logger.warning('No files found for pattern ' + src_pattern)
-            exit_status += 1
+            if output_record.get('required', True):
```

Comment:
This add capability to have output file referenced in the yaml file but will not yield an error or change the exit status if absent